packages/editor/src/client/routes/commentList.tsx

```diff
@@ -146,6 +149,19 @@ export function CommentList() {
 
   const [isConfirmationDialogOpen, setConfirmationDialogOpen] = useState(false)
   const [currentComment, setCurrentComment] = useState<Comment>()
+
+  const fetchParentComment = useCommentQuery({
+    variables: {id: currentComment?.parentID}
+  }).data?.comment?.revisions[length - 1]
+
+  const [currentCommentParent, setCurrentCommentParent] = useState<CommentRevision | undefined>(
+    fetchParentComment
+  )
+
+  useEffect(() => {
```

Comment:
Have you tried fetching together with the comments list? what do you think?

Reply:
@bishoymelek how would that work? We don't know which parent comment to fetch without the user clicking on a comment.

Reply:
If we added resolve function to parentID, it should work fine.

Reply:
You might be right. @penk85 and myself already discussed this. I think there are some circular reference issues. So for now this is the easiest solution.

Reply:
@bishoymelek there were issues with resolving a comment inside a comment, and I haven't found a way around that.

Reply:
@penk85 

file:packages/api/src/graphql/comment.ts
line:118
`export const GraphQLComment: GraphQLObjectType<Comment, Context> = new GraphQLObjectType<
  Comment,
  Context
>({
  name: 'Comment',
  fields: () => ({
    id: {type: GraphQLNonNull(GraphQLID)},

    user: {
      type: GraphQLNonNull(GraphQLUser),
      resolve: createProxyingResolver(({userID}, _, {dbAdapter}) => {
        return dbAdapter.user.getUserByID(userID)
      })
    },
    authorType: {type: GraphQLNonNull(GraphQLCommentAuthorType)},

    itemID: {type: GraphQLNonNull(GraphQLID)},
    itemType: {
      type: GraphQLNonNull(GraphQLCommentItemType)
    },

    parentComment: {
      type: GraphQLComment,
      resolve: createProxyingResolver(({parentID}, _, {dbAdapter}) => {
        if (parentID) return dbAdapter.comment.getCommentById(parentID)
        return null
      })
    },

    revisions: {type: GraphQLNonNull(GraphQLList(GraphQLNonNull(GraphQLCommentRevision)))},

    state: {type: GraphQLNonNull(GraphQLCommentState)},

    rejectionReason: {type: GraphQLCommentRejectionReason},

    createdAt: {type: GraphQLNonNull(GraphQLDateTime)},
    modifiedAt: {type: GraphQLNonNull(GraphQLDateTime)}
  })
})
`